alita_sdk/tools/slack/api_wrapper.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from typing import Optional
+from pydantic import BaseModel, Field, SecretStr, create_model, model_validator
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+from alita_sdk.tools.elitea_base import BaseToolApiWrapper
+
+
+logger = logging.getLogger(__name__)
+
+SendMessageModel = create_model(
+                    "SendMessageModel",
+                    slack_token=(SecretStr, Field( description="Slack Bot/User OAuth Token like XOXB-*****-*****-*****-*****")),
+                    channel_id=(str, Field(description="Channel ID, user ID, or conversation ID to send the message to. (like C12345678 for public channels, D12345678 for DMs)")),
+                    message=(str, Field(description="The message text to send."))
+                    )
+
+ReadMessagesModel = create_model(
+                    "ReadMessagesModel",
+                    slack_token=(SecretStr, Field( description="Slack Bot/User OAuth Token like XOXB-*****-*****-*****-*****")),
+                    channel_id=(str, Field( description="Channel ID, user ID, or conversation ID to read messages from.")),
+                    limit=(int, Field(default=10, description="The number of messages to fetch (default is 10)."))
+                    )
+
+CreateChannelModel = create_model(
+                    "CreateChannelModel",  
+                    slack_token=(SecretStr, Field(description="Slack Bot/User OAuth Token like XOXB-*****-*****-*****-*****")),
+                    channel_name=(str, Field(description="Channel ID, user ID, or conversation ID to send the message to. (like C12345678 for public channels, D12345678 for DMs)")),
+                    is_private=(bool, Field(default=False, description="Whether to make the channel private (default: False)."))
+                    )
+
+ListUsersModel = create_model(
+                    "ListUsersModel",
+                    slack_token=(SecretStr, Field(description="Slack Bot/User OAuth Token like XOXB-*****-*****-*****-*****"))
+                    )
+
+
+class SlackApiWrapper(BaseToolApiWrapper):
+
+    """
+    Slack API wrapper for interacting with Slack channels and messages.
+    """
+    
+    def send_message(self, slack_token : SecretStr, channel_id: str, message: str):
```

Comment:
No need to pass slack_token and chanell_id to all methods.
They could be available on SlackApiWrapper instance level (if you declare class atributes).
They are passing as kwargs in "slack_api_wrapper = SlackApiWrapper(**kwargs)"
Example: https://github.com/ProjectAlita/alita-sdk/blob/5e2a5b55acc0095e1133e82d8049b620094c28c5/alita_sdk/tools/figma/api_wrapper.py#L230